app/services/data_sources/json_user_data_source.rb

```diff
@@ -146,14 +146,12 @@ def update_user!(user: nil)
     end
 
     def delete_user!(user: nil)
-      user_index, existing_user = find_user_index_and_existing_user(user: user)
```

Comment:
From what I understand, we can remove this function completely, once we also update `update_user!` or anywhere else where we use `find_user_index_and_existing_user` 👍

Reply:
Hey @KrauseFx , I updated the `update_user` method, and added specs for it. Could you let me know how you feel about the new implementation? Thanks! 💯